pkg/image/api/helper.go

```diff
@@ -642,7 +642,8 @@ func DifferentTagEvent(stream *ImageStream, tag string, next TagEvent) bool {
 	previous := &tags.Items[0]
 	sameRef := previous.DockerImageReference == next.DockerImageReference
 	sameImage := previous.Image == next.Image
-	return !(sameRef && sameImage)
+	sameGeneration := previous.Generation == next.Generation
+	return !(sameRef && sameImage && sameGeneration)
```

Comment:
Not sure - there was a reason not to mutate the generation, so knowing which callers depended on it and which use cases would help clarify.  I'm scared there aren't enough tests to be sure.